src/sync/chase_lev.rs

```diff
@@ -158,69 +155,93 @@ impl<T> Deque<T> {
     unsafe fn push(&self, data: T) {
         let guard = epoch::pin();
 
-        let mut b = self.bottom.load(SeqCst);
-        let t = self.top.load(SeqCst);
-        let mut a = self.array.load(SeqCst, &guard).unwrap();
-        let size = b - t;
+        let mut b = self.bottom.load(Relaxed);
+        let t = self.top.load(Acquire);
+        let mut a = self.array.load(Relaxed, &guard).unwrap();
+        
+        let size = b.wrapping_sub(t);
```

Comment:
Also: the two are equivalent in this case, since we first decrement b and then compare it to t. If they are equal we can store b+1 or t+1 at the end, it makes no difference.